packages/server/src/fhir/repo.ts

```diff
@@ -1702,42 +1702,60 @@ export class Repository extends FhirRepository<PoolClient> implements Disposable
    * Returns the author reference string (resourceType/id).
    * If the current context is a ClientApplication, handles "on behalf of".
    * Otherwise uses the current context profile.
-   * @param existing - Existing resource if one exists.
+   * @param existing - Current (soon to be previous) resource, if one exists.
    * @param updated - The incoming updated resource.
-   * @param create - Flag for when "creating" vs "updating".
-   * @returns The account value.
-   */
-  private async getAccounts(
-    existing: Resource | undefined,
-    updated: Resource,
-    create: boolean
-  ): Promise<Reference[] | undefined> {
+   * @returns The account values.
+   */
+  private async getAccounts(existing: Resource | undefined, updated: Resource): Promise<Reference[] | undefined> {
     const updatedAccounts = this.extractAccountReferences(updated.meta);
     if (updatedAccounts && this.canWriteAccount()) {
       // If the user specifies an account, allow it if they have permission.
       return updatedAccounts;
     }
 
-    if (create && this.context.accessPolicy?.compartment) {
-      // If the user access policy specifies a compartment, then use it as the account.
-      return [this.context.accessPolicy.compartment];
+    const accounts = new Set<string>();
+    if (!existing && this.context.accessPolicy?.compartment?.reference) {
+      // If the creator's access policy specifies a compartment, then use it as the account.
+      // The writer's access policy is only applied at resource creation: simply editing a
+      // resource does NOT pull it into the user's account.
+      accounts.add(this.context.accessPolicy.compartment.reference);
     }
 
-    if (updated.resourceType !== 'Patient') {
+    if (updated.resourceType === 'Patient') {
+      // Otherwise, default to the existing value.
```

Comment:
Minor nit, I think this comment is a copy/paste artifact.

```suggestion
      // When examining a Patient resource, we only look at the individual patient
      // We should not call `getPatients` and `readReference`
```